projects/storefrontlib/cms-components/myaccount/consent-management/components/consent-management.component.ts

```diff
@@ -276,8 +281,8 @@ export class ConsentManagementComponent implements OnInit, OnDestroy {
         if (this.isRequiredConsent(template)) {
           return;
         }
+        consentsToGive.push(template);
       }
-      consentsToGive.push(template);
     });
```

Comment:
The closing parenthesis `}` of the `if` statement has been moved. I believe now the whole function `showAll` has a slightly different logic than original.  

Was it a typo? Or it was a deliberate change?
If deliberate, why it was an important change for you? Was it needed for CDC only or also it was fixing something in the core feature?

(it might be perceived as a behavioral breaking change. So we would need to justify it very well, if we want to release this breaking change in a minor version. That's why I need to understand it well)

Reply:
All required consents will be disabled in the UI. When we do a 'select All' in my-account->consent management, ideally these disabled consents should not be picked for updated. This is done correctly in 'rejectAll' method, so I perceived this as a bug and fixed it here

Reply:
Understood. Code-wise it seems to be a justified bugfix.

For the clarity of the git history: what do you think about reverting this change from this PR, and creating a new `fix: ...` PR as a followup of this PR (+ it would be good to add unit test for it, because current tests didn't spot this bug)?